### ISSUE TITLE
Enhancement: Icon color health bar

### ISSUE THREAD

**Issue by CEliuxJV** [FunkinCrew/Funkin#3129]
### Issue Checklist

- [X] I have properly named the enhancement
- [X] I have checked the issues/discussions pages to see if the enhancement has been previously suggested

### What is your suggestion, and why should it be implemented?

I know that this is something that differentiates the base game from the other engines, however I consider that it is something that makes the game's hud look better and that the colors of the icons do not clash with the colors of the bar of health.

I think adding an option to choose whether you want the health bar to have the default red and green colors or the colors of the character icons would be a good addition, as well as it would be an option that can only be activated by people who want to see it, and those who want to leave the base game style of the health bar just leave the option disabled.
![screenshot-2024-08-29-18-15-31](https://github.com/user-attachments/assets/4eed43c8-7c38-443e-8648-d300a91fe66f)

![Captura de pantalla 2024-08-29 181415](https://github.com/user-attachments/assets/838ddb3b-c335-4239-8347-e7a553a7e1d0)

Since the life bar icon information is saved in the same character .json file, an option could be added to the animation editor to choose the color of its side of the life bar, and that the color id is saved in the character's .json, just as the Psych Engine does.
![Captura de pantalla 2024-08-29 181927](https://github.com/user-attachments/assets/36302315-090b-402a-8f8f-0c98e6fd2a0c)

Although I consider that a good implementation to complement this idea and not just be a copy of the Psych Engine would be to add a dropper to more easily choose the appropriate color of the health bar and not have to use another external application.

Also another good implementation would be to add an offset modifier for the icons, since I notice that the icons of some characters, having low life, have a different position than the normal icon, and that some are very close to the BF icon. This is not something that can be easily noticed, but when you notice it you cannot unsee it.
https://github.com/user-attachments/assets/eeb7aae0-dc27-4fb7-8b07-9a2a1010e71c

Also an extra improvement to the health bar is the request that was made in the discussions part about 3 years ago #1628, although I don't know if anyone else has proposed it, I think that the transition of gradually lowering the life and score of the previous song over the weeks would be very good.

Psdt:
Sorry if something is not understood well, I wrote this with the translator.

**Comment by amyspark-ng** [comment]
if the icons will be kept having a single color, instead of a dropper or storing the icon color separetely, the color could just be picked as the average of the icon, this would also mix all the colors for example in daddy and mommy on week 5 or pump and skid on week 2 but i don't know how good that'd look

**Comment by CEliuxJV** [comment]
> if the icons will be kept having a single color, instead of a dropper or storing the icon color separetely, the color could just be picked as the average of the icon, this would also mix all the colors for example in daddy and mommy on week 5 or pump and skid on week 2 but i don't know how good that'd look

I understand what you are saying, however choosing the average color of the icon could make the black color of the borders combine with the color of the icon itself, also choosing the color with a dropper or with the color id and saving it in the . json of the character can help choose the exact color or choose another color that is not necessarily the one that is most present. An example of the latter is in the colors of the icon bar in Twinsomnia, which do not use the most abundant color of the icon:

![Captura de pantalla 2024-09-10 232450](https://github.com/user-attachments/assets/4ece9274-b4e1-42a5-be7a-ee937d5c4470)

Or the color of Tankman's life bar, because the life bar being completely black can look strange:

![Tankman Example](https://github.com/user-attachments/assets/2d8a1b47-8a51-4c88-9dd0-d874e9660327)

I also think that making the game choose average color automatically may be a bit more complicated to do.

I also mixed the colors of the parents and Skid and Pump and they don't look too bad, so I guess icons with very different colors don't look bad and wouldn't be a big problem.

![Skid   Pump and Parents example](https://github.com/user-attachments/assets/5ead50a1-33b5-465d-af7c-1bbac37b5719)

**Comment by CEliuxJV** [comment]
Since they released 0.5 with new character animations, when they release an update and fix the animation editor, I will update the .zip with the .json with the new sprites and character animations with the offsets fixed

Psdt: Funkin' Crew, please let me offset the character animations, I'm a perfectionist.
And if I get over how, I fix the offsets of the other animations, like BF and Pico in Freeplay, in the character selector and in the week menu.

**Comment by RachRach696** [comment]
I also suggest it would affect the beta icon for when you press 9, the player health bar color will automatically become whatever the main color for the beta icon for the player would be.

**Comment by CEliuxJV** [comment]
> I also suggest it would affect the beta icon for when you press 9, the player health bar color will automatically become whatever the main color for the beta icon for the player would be.

True, I had not thought about that, but yes, by having the icon color health bar option activated, when you press 9 to change the icon to that of BF Beta, the color of the life bar changes along with with the icon so that they are not different colors.

This could be added next to the lines of code that change the icon to BF Beta.
The BF Beta icon color is in RGB: 233, 255, 72. Or in HEX: #e9ff48
